library/Icinga/Application/Hook/Common/DbMigration.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/* Icinga Web 2 | (c) 2023 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Application\Hook\Common;
+
+use ipl\Sql\Connection;
+use RuntimeException;
+
+class DbMigration
+{
+    /** @var string The sql string to be executed */
+    protected $query;
+
+    /** @var string The sql script version the queries are loaded from */
+    protected $version;
+
+    /** @var string */
+    protected $scriptPath;
+
+    /** @var ?string */
+    protected $description;
+
+    /** @var ?string */
+    protected $lastState;
+
+    public function __construct(string $version, string $scriptPath)
+    {
+        $this->scriptPath = $scriptPath;
+
+        $this->setVersion($version);
+    }
+
+    /**
+     * Get the sql script version the queries are loaded from
+     *
+     * @return string
+     */
+    public function getVersion(): string
+    {
+        return $this->version;
+    }
+
+    /**
+     * Set the sql script version the queries are loaded from
+     *
+     * @param string $version
+     *
+     * @return $this
+     */
+    public function setVersion(string $version): self
+    {
+        $this->version = $version;
+
+        return $this;
+    }
+
+    /**
+     * Get upgrade script relative path name
+     *
+     * @return string
+     */
+    public function getScriptPath(): string
+    {
+        return $this->scriptPath;
+    }
+
+    /**
+     * Get the description of this database migration if any
+     *
+     * @return ?string
+     */
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    /**
+     * Set the description of this database migration
+     *
+     * @param ?string $description
+     *
+     * @return DbMigration
+     */
+    public function setDescription(?string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    /**
+     * Get the last error message of this hook if any
+     *
+     * @return ?string
+     */
+    public function getLastState(): ?string
+    {
+        return $this->lastState;
+    }
+
+    /**
+     * Set the last error message
+     *
+     * @param ?string $message
+     *
+     * @return $this
+     */
+    public function setLastState(?string $message): self
+    {
+        $this->lastState = $message;
+
+        return $this;
+    }
+
+    /**
+     * Perform the sql migration
+     *
+     * @param Connection $conn
+     *
+     * @return $this
+     *
+     * @throws RuntimeException Throws an error in case of any database errors or when there is nothing to migrate
+     */
+    public function apply(Connection $conn): self
+    {
+        if (! $this->query) {
+            $statements = @file_get_contents($this->getScriptPath());
+            if ($statements === false) {
+                throw new RuntimeException(sprintf('Cannot load upgrade script %s', $this->getScriptPath()));
+            }
+
+            if (empty($this->query)) {
```

Comment:
you haven't tested this, I suppose

Reply:
> you haven't tested this, I suppose

I don't know what's this supposed to say!
![Bildschirmfoto 2023-09-12 um 17 56 06](https://github.com/Icinga/icingaweb2/assets/57616252/311f80cd-8f90-4155-95a4-804c33bb5381)

Reply:
<img width="624" alt="Bildschirmfoto 2023-09-12 um 17 57 07" src="https://github.com/Icinga/icingaweb2/assets/57616252/6f2d9538-41cb-4aaf-aadd-896ae7c37880">

Reply:
🤦‍♂️